static/aggregate_test.go

```diff
@@ -0,0 +1,262 @@
+package static_test
+
+import (
+	"github.com/dogmatiq/configkit"
+	"github.com/dogmatiq/configkit/internal/entity"
+	"github.com/dogmatiq/configkit/message"
+	. "github.com/dogmatiq/configkit/static"
+	"github.com/dogmatiq/dogma/fixtures"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"golang.org/x/tools/go/packages"
+)
+
+var _ = Describe("func FromPackages() (aggregate analysis)", func() {
+	When("the application contains a single aggregate handler", func() {
+		It("returns the aggregate handler configuration", func() {
+			cfg := packages.Config{
+				Mode: packages.LoadAllSyntax,
+				Dir:  "testdata/aggregates/single-aggregate-app",
+			}
+
+			pkgs, err := packages.Load(&cfg, "./...")
+			Expect(err).NotTo(HaveOccurred())
+
+			apps := FromPackages(pkgs)
+
+			Expect(apps).To(
+				ConsistOf(
+					PointTo(
```

Comment:
These tests have made me aware of how dependent they are on the internal fields of the entity structs. It might be really tricky to update these tests if the implementation of those entity structs were to change. 

I think we should spend some time to workout how we can test this via the `configkit` interfaces, without expecting a specific implementation. 

In addition to that I think it would be good to focus the expectations on the handler configs, and less so on the application in an effort to reduce the cognitive overhead of working out what went wrong when one of the expectations does fail.

Reply:
Simplified the tests as per the suggestion above in https://github.com/dogmatiq/configkit/pull/123/commits/2d1ce67c9a4b0f59e72e72a333aaccf30496e10f.  The first single aggregate handler test still uses `github.com/onsi/gomega/gstruct` assertions because of multiple messages for each type in `message.NameRoles` maps. All remaining tests are now using direct `Equal()` assertions.

Reply:
Another go in https://github.com/dogmatiq/configkit/pull/123/commits/23b4fd9461639f5459c139229b0f847640b4af82 on simplifying the aggregate tests as we agreed during our recent talk. This time aI am making sure that we are not using any internal types for assertions.